Scripts/RemediationScripts/Remediate-AddDiagnosticSettingsForLogicApps.ps1

```diff
@@ -0,0 +1,596 @@
+
+        # Get all Logic Apps in a Subscription.
+        $logicApps = Get-AzResource -ResourceType "Microsoft.Logic/workflows" -ErrorAction Stop -ResourceGroupName "v-abbhanTestRG"
+        
+        $logicAppResources += $logicApps | Select-Object @{N='ResourceId';E={$_.ResourceId}},
+                                                                     @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                     @{N='Name';E={$_.ResourceName}},
+                                                                     @{N='ResourceType';E={$_.ResourceType}},
+                                                                     @{N='Location';E={$_.Location}}
+    }
+    else
+    {
+        if (-not (Test-Path -Path $FilePath))
+        {
+            Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+
+        Write-Host "Fetching all Logic Apps from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+
+        # Importing the list of Logic Apps to be remediated.
+        $logicAppDetails = Import-Csv -LiteralPath $FilePath
+
+        $logicAppResources = $logicAppDetails | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+    }
+
+    $totalLogicApps = $logicAppResources.Count
+
+    if ($totalLogicApps -eq 0)
+    {
+        Write-Host "No Logic Apps found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "Found $($totalLogicApps) Logic Apps." -ForegroundColor $([Constants]::MessageType.Update)
+
+    # Includes Logic Apps where required diagnostic settings are present.
+    $logicAppsWithRequiredDiagnosticSetting = @()
+
+    # Includes Logic Apps where required diagnostic settings are not present.
+    $logicAppsWithoutRequiredDiagnosticSetting = @()
+
+  # Check if required Diagnostic setting for Logic App is present
+
+    $logicAppResources | ForEach-Object {
+        try
+        {
+            Write-Host "Fetching diagnostic settings for Resource ID - $($_.ResourceId)"
+
+            $settings= Get-AzDiagnosticSetting -ResourceId $_.ResourceId
+            
+            if ($null -eq $settings)
+            {
+                # No diagnostic setting is configured
+                $logicAppsWithoutRequiredDiagnosticSetting += $_
+            }
+            #TODO: Add CategoryGroup Based check 
+            else{
+                if(($settings.Log|Where-Object {$_.Category -eq "WorkflowRuntime" -and $_.Enabled -eq "True"}|Measure-Object).Count -gt 0 ){
+                    $logicAppsWithRequiredDiagnosticSetting += $_
+                }
+                else{
+                    $logicAppsWithoutRequiredDiagnosticSetting += $_
+                }
+            }
+        }
+        catch
+        {
+            $logicAppsWithoutRequiredDiagnosticSetting += $_ 
```

Comment:
If any error occurs while fetching the Diagnostic settings, then should we consider withoutRequiredDiagnostic Settings or skip that resource?

Reply:
Adding one more diagnostic setting will not harm much. To cover max resources added those to without diagn settings

Reply:
IF not too hard, better to skip.